src/server/api/AdminPanel.ts

```diff
@@ -0,0 +1,208 @@
+		return;
+	}
+	const reason = commandAndArgs[2];
+	const username = commandAndArgs[1];
+	const userid = getMemberDataByCriteria(["user_id"], "username", username)["user_id"];
+	if (userid !== undefined) {
+		deleteAccount(userid, reason ?? "");
+	}
+	else {
+		res.status(404).send("User " + username + " does not exist.");
+		return;
+	}
+	res.status(200).send("Successfully deleted user " + username + ".");
+}
+
+function usernameCommand(command: string, commandAndArgs: string[], res: Response) {
+	if (commandAndArgs[1] === "get") {
+		if (commandAndArgs.length < 3) {
+			res.status(422).send("Invalid number of arguments, expected 2, got " + (commandAndArgs.length - 1) + ".");
+			return;
+		}
+		const parsedId = Number.parseInt(commandAndArgs[2]!);
+		if (Number.isNaN(parsedId)) {
+			res.status(422).send("User id must be an integer.");
+			return;
+		}
+		const username = getMemberDataByCriteria(["username"], "user_id", parsedId)["username"];
+		logEvents("Command executed: " + command + "\nResult: " + username + "\n", "adminCommands");
+		if (username === undefined) {
+			res.status(404).send("User with id " + parsedId + " does not exist.");
+		}
+		else {
+			res.status(200).send(username);
+		}
+	}
+	else if (commandAndArgs[1] === "set") {
+		if (commandAndArgs.length < 4) {
+			res.status(422).send("Invalid number of arguments, expected 3, got " + (commandAndArgs.length - 1) + ".");
+			return;
+		}
+		// TODO add username changing logic
+		res.status(503).send("Changing usernames is not yet supported.");
+	}
+	else if (commandAndArgs[1] === undefined) {
+		res.status(422).send("Expected either get or set as a subcommand.");
+	}
+	else {
+		res.status(422).send("Invalid subcommand, expected either get or set, got " + commandAndArgs[1] + ".");
+	}
+}
+
+function logoutUser(command: string, commandAndArgs: string[], res: Response) {
+	if (commandAndArgs.length < 2) {
+		res.status(422).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+		return;
+	}
+	logCommand(command);
+	const username = commandAndArgs[1];
+	const userid = getMemberDataByCriteria(["user_id"], "username", username)["user_id"];
+	if (userid !== undefined) {
+		deleteAllSessionsOfUser(userid);
+	}
+	else {
+		res.status(404).send("User " + username + " does not exist.");
+	}
+	res.status(200).send("User " + username + " successfully logged out.");
+}
+
+function getUserInfo(command: string, commandAndArgs: string[], res: Response) {
+	if (commandAndArgs.length < 2) {
+		res.status(422).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+		return;
+	}
+	const username = commandAndArgs[1];
+	const memberInfo = getMemberDataByCriteria(["user_id", "username", "roles", "joined", "last_seen", "preferences", "verification", "username_history"],
+		"username",
+		username);
+	logEvents("Command executed: " + command + "\n" + memberInfo + "\n", "adminCommands");
```

Comment:
Perhaps add "Result: " to the second line to be consistent when logging commands that include a result?